lib/resolvers/http.js

```diff
@@ -99,16 +99,29 @@ async function download (u, httpOptions, redirects) {
 
   redirects.push(u.href);
 
+  // Certain implementations of `fetch` do not support URL based basic-auth
+  // Convert to headers
+  const headers = new Headers(httpOptions.headers);
```

Comment:
this is the actual fix

Reply:
This would probably make a good named function, `moveBasicAuthFromUrlToHeader(u, httpOptions.headers): { url, headers }` or something like that.